blsSignatures/blsSignatures.go

```diff
@@ -177,41 +192,42 @@ func VerifyAggregatedSignatureDifferentMessages(sig Signature, messages [][]byte
 //
 // If keyValidationMode is true, this uses a tweaked version of the padding,
 // so that the result will not collide with a result generated in an ordinary signature.
-func hashToG1Curve(message []byte, keyValidationMode bool) (*bls12381.PointG1, error) {
+func hashToG1Curve(message []byte, keyValidationMode bool) (*bls12381.G1Affine, error) {
 	var padding [16]byte
 	h := crypto.Keccak256(message)
 	if keyValidationMode {
 		// modify padding, for domain separation
 		padding[0] = 1
 	}
-	g1 := bls12381.NewG1()
-	return g1.MapToCurve(append(padding[:], h...))
+	fp := new(fp.Element)
+	fp.Unmarshal(append(padding[:], h...))
+	res := bls12381.MapToG1(*fp)
+	return &res, nil
 }
 
 func PublicKeyToBytes(pub PublicKey) []byte {
-	g2 := bls12381.NewG2()
+	keyBytes := pub.key.Bytes()
```

Comment:
We should add a test with a hardcoded public key generated from the old version that deserializing it and then serializing it results in the same thing.

Reply:
Done, covered in TestPublicKeyToBytes.